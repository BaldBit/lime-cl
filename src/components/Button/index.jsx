import React from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash-es/noop';

import ButtonContainer from './styled';

const Button = (props) => {
  const { value, mode, size, disabled, loading, fluid, onClick, ...otherProps } = props;

  return (
    <ButtonContainer {...otherProps} mode={mode} size={size} disabled={disabled} fluid={fluid} onClick={onClick}>
      {value}
    </ButtonContainer>
  );
};

Button.propTypes = {
  value: PropTypes.string.isRequired,
  mode: PropTypes.oneOf(['primary', 'secondary']),
  size: PropTypes.oneOf(['regular', 'big']),
  disabled: PropTypes.bool,
  loading: PropTypes.bool,
  fluid: PropTypes.bool,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  mode: 'primary',
  size: 'regular',
  disabled: false,
  loading: false,
  fluid: false,
  onClick: noop,
};

export default Button;
