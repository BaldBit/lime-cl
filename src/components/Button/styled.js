import styled from 'styled-components';
import { darken } from 'polished';

import limeLight from '../Theme/limeLight';

/**
 * get color mode for the button based on the disabled prop
 *
 * @param {boolean} disabled
 */
const getColor = (disabled) => (disabled ? limeLight.colorGrayMedium : limeLight.colorPrimaryOne);

/**
 * Set button's background, border and font color based on mode. If user didn't specify any value for the
 * mode it will used neutral color similar to a disable button's look and feel
 *
 * @param {string} [mode=primary] - there are three modes; 'primary', 'secondary' and 'null' value
 * @param {boolean} disabled - set disable colors if the button is disabled
 */
const getButtonMode = (mode, disabled) => {
  let css = '';

  if (mode === 'primary') {
    // primary mode style
    css = `
      background-color: ${getColor(disabled)};
      border-color: ${getColor(disabled)};
      color: ${limeLight.colorLight};

      ${
        !disabled
          ? `
        &:hover {
          background-color: ${darken(0.05, limeLight.colorPrimaryOne)};
          border-color: ${darken(0.05, limeLight.colorPrimaryOne)};
        }
      `
          : ''
      }
    `;
  } else if (mode === 'secondary') {
    // secondary mode style
    css = `
      border-color: ${getColor(disabled)};
      color: ${getColor(disabled)};

      ${
        !disabled
          ? `
        &:hover {
          border-color: ${darken(0.2, limeLight.colorPrimaryOne)};
          color: ${darken(0.2, limeLight.colorPrimaryOne)};
        }
      `
          : ''
      }
    `;
  } else {
    // neutral mode style
    css = `
      border-color: ${getColor(true)};
      color: ${getColor(true)};

      &:hover {
        border-color: ${darken(0.2, getColor(true))};
        color: ${darken(0.2, getColor(true))};
      }
    `;
  }

  return css;
};

const ButtonContainer = styled.button`
  display: inline-flex;
  border-width: 1px;
  border-style: solid;
  cursor: pointer;
  outline: 1px dotted transparent;
  outline-offset: 2px;
  justify-content: center;
  align-items: center;
  transition: 0.3s border-color, 0.3s background-color, 0.3s color, 0.3s outline-color;

  &:focus {
    outline-color: ${limeLight.colorGrayDark};
  }

  ${({ size, fluid, mode, disabled }) => `
    font-size: ${limeLight.fontSize[size]};
    padding: ${limeLight.button.padding[size]};
    ${fluid ? 'width: 100%' : ''};
    border-radius: ${limeLight.button.borderRadius[size]};
    ${getButtonMode(mode, disabled)};
  `}
`;

const LoaderContainer = styled.span`
  margin-right: 5px;
`;

export { ButtonContainer, LoaderContainer };
