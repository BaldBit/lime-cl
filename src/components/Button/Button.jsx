import React from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash-es/noop';

import Loader from '../Loader/Loader';
import { ButtonContainer, LoaderContainer } from './styled';

const Button = (props) => {
  const { value, mode, size, disabled, loading, fluid, onClick, ...otherProps } = props;

  return (
    <ButtonContainer {...otherProps} mode={mode} size={size} disabled={disabled} fluid={fluid} onClick={onClick}>
      {loading && (
        <LoaderContainer>
          <Loader size={size} />
        </LoaderContainer>
      )}
      {value}
    </ButtonContainer>
  );
};

Button.propTypes = {
  /**
   * @property {string} value - Text to display on the button element
   */
  value: PropTypes.string.isRequired,
  /**
   * Text to display on the button element
   * @property {string} [mode=primary] - Set style of the button
   */
  mode: PropTypes.oneOf(['primary', 'secondary', 'neutral']),
  /**
   * Text to display on the button element
   * @property {string} [size=regular] - Set the size of the button
   */
  size: PropTypes.oneOf(['regular', 'big']),
  /**
   * @property {boolean} [disabled=false] - Set the disable property of the button element
   */
  disabled: PropTypes.bool,
  /**
   * @property {boolean} [loading=false] - Show the Loading component
   */
  loading: PropTypes.bool,
  /**
   * @property {boolean} [fluid=false] - Stretch the button to it's parent's width
   */
  fluid: PropTypes.bool,
  /**
   * @param {Button~buttonOnClickCallback} onClick - The callback that invoke when click on the button
   */
  onClick: PropTypes.func,
};

Button.defaultProps = {
  mode: 'primary',
  size: 'regular',
  disabled: false,
  loading: false,
  fluid: false,
  onClick: noop,
};

export default Button;

/**
 * Button click callback
 * @callback buttonOnClickCallback
 * @param {object} event
 */
