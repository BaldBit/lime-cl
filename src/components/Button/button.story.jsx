import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, boolean, select } from '@storybook/addon-knobs';

import Button from './Button';

const stories = storiesOf('Button', module);

stories.addDecorator(withKnobs);

stories.add('Default', () => (
  <Button
    value={text('Value', 'Lime Button')}
    mode={select('Mode', ['primary', 'secondary', 'neutral'], 'primary')}
    size={select('Size', ['regular', 'big'], 'regular')}
    disabled={boolean('Disabled', false)}
    loading={boolean('Loading', false)}
    fluid={boolean('Fluid', false)}
    onClick={action('button-click')}
  />
));
