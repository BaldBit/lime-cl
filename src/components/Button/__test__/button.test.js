import React from 'react';
import { mount } from 'enzyme';

import Button from '../Button';

it('renders button with primary mode', () => {
  const wrapper = mount(<Button value="button" />);
  expect(wrapper).toMatchSnapshot();
});

it('Render button with primary mode in disabled state', () => {
  const wrapper = mount(<Button value="button" disabled />);
  expect(wrapper).toMatchSnapshot();
});

it('Render button with secondary mode', () => {
  const wrapper = mount(<Button value="button" mode="secondary" />);
  expect(wrapper).toMatchSnapshot();
});

it('Render button with secondary mode in disabled state', () => {
  const wrapper = mount(<Button value="button" mode="secondary" disabled />);
  expect(wrapper).toMatchSnapshot();
});

it('Render button without a mode (neutral)', () => {
  const wrapper = mount(<Button value="button" mode="neutral" />);
  expect(wrapper).toMatchSnapshot();
});

it('Render button with fluid prop', () => {
  const wrapper = mount(<Button value="button" fluid />);
  expect(wrapper).toMatchSnapshot();
});

it('Render button Loader component', () => {
  const wrapper = mount(<Button value="button" loading />);
  expect(wrapper.find('Loader')).toHaveLength(1);
});
