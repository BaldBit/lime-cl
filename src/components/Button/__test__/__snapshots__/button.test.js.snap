// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render button with fluid prop 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-width: 1px;
  border-style: solid;
  cursor: pointer;
  outline: 1px dotted transparent;
  outline-offset: 2px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: 0.3s border-color,0.3s background-color,0.3s color,0.3s outline-color;
  transition: 0.3s border-color,0.3s background-color,0.3s color,0.3s outline-color;
  font-size: 16px;
  padding: 8px;
  width: 100%;
  border-radius: 3px;
  background-color: #2ace7e;
  border-color: #2ace7e;
  color: #ffffff;
}

.c0:focus {
  outline-color: #6A737B;
}

.c0:hover {
  background-color: #26b971;
  border-color: #26b971;
}

<Button
  disabled={false}
  fluid={true}
  loading={false}
  mode="primary"
  onClick={[Function]}
  size="regular"
  value="button"
>
  <styled.button
    disabled={false}
    fluid={true}
    mode="primary"
    onClick={[Function]}
    size="regular"
  >
    <StyledComponent
      disabled={false}
      fluid={true}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: inline-flex;
  border-width: 1px;
  border-style: solid;
  cursor: pointer;
  outline: 1px dotted transparent;
  outline-offset: 2px;
  justify-content: center;
  align-items: center;
  transition: 0.3s border-color, 0.3s background-color, 0.3s color, 0.3s outline-color;

  &:focus {
    outline-color: ",
              "#6A737B",
              ";
  }

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      mode="primary"
      onClick={[Function]}
      size="regular"
    >
      <button
        className="c0"
        disabled={false}
        mode="primary"
        onClick={[Function]}
        size="regular"
      >
        button
      </button>
    </StyledComponent>
  </styled.button>
</Button>
`;

exports[`Render button with primary mode in disabled state 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-width: 1px;
  border-style: solid;
  cursor: pointer;
  outline: 1px dotted transparent;
  outline-offset: 2px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: 0.3s border-color,0.3s background-color,0.3s color,0.3s outline-color;
  transition: 0.3s border-color,0.3s background-color,0.3s color,0.3s outline-color;
  font-size: 16px;
  padding: 8px;
  border-radius: 3px;
  background-color: #A1A1A1;
  border-color: #A1A1A1;
  color: #ffffff;
}

.c0:focus {
  outline-color: #6A737B;
}

<Button
  disabled={true}
  fluid={false}
  loading={false}
  mode="primary"
  onClick={[Function]}
  size="regular"
  value="button"
>
  <styled.button
    disabled={true}
    fluid={false}
    mode="primary"
    onClick={[Function]}
    size="regular"
  >
    <StyledComponent
      disabled={true}
      fluid={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: inline-flex;
  border-width: 1px;
  border-style: solid;
  cursor: pointer;
  outline: 1px dotted transparent;
  outline-offset: 2px;
  justify-content: center;
  align-items: center;
  transition: 0.3s border-color, 0.3s background-color, 0.3s color, 0.3s outline-color;

  &:focus {
    outline-color: ",
              "#6A737B",
              ";
  }

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      mode="primary"
      onClick={[Function]}
      size="regular"
    >
      <button
        className="c0"
        disabled={true}
        mode="primary"
        onClick={[Function]}
        size="regular"
      >
        button
      </button>
    </StyledComponent>
  </styled.button>
</Button>
`;

exports[`Render button with secondary mode 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-width: 1px;
  border-style: solid;
  cursor: pointer;
  outline: 1px dotted transparent;
  outline-offset: 2px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: 0.3s border-color,0.3s background-color,0.3s color,0.3s outline-color;
  transition: 0.3s border-color,0.3s background-color,0.3s color,0.3s outline-color;
  font-size: 16px;
  padding: 8px;
  border-radius: 3px;
  border-color: #2ace7e;
  color: #2ace7e;
}

.c0:focus {
  outline-color: #6A737B;
}

.c0:hover {
  border-color: #19794a;
  color: #19794a;
}

<Button
  disabled={false}
  fluid={false}
  loading={false}
  mode="secondary"
  onClick={[Function]}
  size="regular"
  value="button"
>
  <styled.button
    disabled={false}
    fluid={false}
    mode="secondary"
    onClick={[Function]}
    size="regular"
  >
    <StyledComponent
      disabled={false}
      fluid={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: inline-flex;
  border-width: 1px;
  border-style: solid;
  cursor: pointer;
  outline: 1px dotted transparent;
  outline-offset: 2px;
  justify-content: center;
  align-items: center;
  transition: 0.3s border-color, 0.3s background-color, 0.3s color, 0.3s outline-color;

  &:focus {
    outline-color: ",
              "#6A737B",
              ";
  }

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      mode="secondary"
      onClick={[Function]}
      size="regular"
    >
      <button
        className="c0"
        disabled={false}
        mode="secondary"
        onClick={[Function]}
        size="regular"
      >
        button
      </button>
    </StyledComponent>
  </styled.button>
</Button>
`;

exports[`Render button with secondary mode in disabled state 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-width: 1px;
  border-style: solid;
  cursor: pointer;
  outline: 1px dotted transparent;
  outline-offset: 2px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: 0.3s border-color,0.3s background-color,0.3s color,0.3s outline-color;
  transition: 0.3s border-color,0.3s background-color,0.3s color,0.3s outline-color;
  font-size: 16px;
  padding: 8px;
  border-radius: 3px;
  border-color: #A1A1A1;
  color: #A1A1A1;
}

.c0:focus {
  outline-color: #6A737B;
}

<Button
  disabled={true}
  fluid={false}
  loading={false}
  mode="secondary"
  onClick={[Function]}
  size="regular"
  value="button"
>
  <styled.button
    disabled={true}
    fluid={false}
    mode="secondary"
    onClick={[Function]}
    size="regular"
  >
    <StyledComponent
      disabled={true}
      fluid={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: inline-flex;
  border-width: 1px;
  border-style: solid;
  cursor: pointer;
  outline: 1px dotted transparent;
  outline-offset: 2px;
  justify-content: center;
  align-items: center;
  transition: 0.3s border-color, 0.3s background-color, 0.3s color, 0.3s outline-color;

  &:focus {
    outline-color: ",
              "#6A737B",
              ";
  }

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      mode="secondary"
      onClick={[Function]}
      size="regular"
    >
      <button
        className="c0"
        disabled={true}
        mode="secondary"
        onClick={[Function]}
        size="regular"
      >
        button
      </button>
    </StyledComponent>
  </styled.button>
</Button>
`;

exports[`Render button without a mode (neutral) 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-width: 1px;
  border-style: solid;
  cursor: pointer;
  outline: 1px dotted transparent;
  outline-offset: 2px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: 0.3s border-color,0.3s background-color,0.3s color,0.3s outline-color;
  transition: 0.3s border-color,0.3s background-color,0.3s color,0.3s outline-color;
  font-size: 16px;
  padding: 8px;
  border-radius: 3px;
  border-color: #A1A1A1;
  color: #A1A1A1;
}

.c0:focus {
  outline-color: #6A737B;
}

.c0:hover {
  border-color: #6e6e6e;
  color: #6e6e6e;
}

<Button
  disabled={false}
  fluid={false}
  loading={false}
  mode="neutral"
  onClick={[Function]}
  size="regular"
  value="button"
>
  <styled.button
    disabled={false}
    fluid={false}
    mode="neutral"
    onClick={[Function]}
    size="regular"
  >
    <StyledComponent
      disabled={false}
      fluid={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: inline-flex;
  border-width: 1px;
  border-style: solid;
  cursor: pointer;
  outline: 1px dotted transparent;
  outline-offset: 2px;
  justify-content: center;
  align-items: center;
  transition: 0.3s border-color, 0.3s background-color, 0.3s color, 0.3s outline-color;

  &:focus {
    outline-color: ",
              "#6A737B",
              ";
  }

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      mode="neutral"
      onClick={[Function]}
      size="regular"
    >
      <button
        className="c0"
        disabled={false}
        mode="neutral"
        onClick={[Function]}
        size="regular"
      >
        button
      </button>
    </StyledComponent>
  </styled.button>
</Button>
`;

exports[`renders button with primary mode 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-width: 1px;
  border-style: solid;
  cursor: pointer;
  outline: 1px dotted transparent;
  outline-offset: 2px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: 0.3s border-color,0.3s background-color,0.3s color,0.3s outline-color;
  transition: 0.3s border-color,0.3s background-color,0.3s color,0.3s outline-color;
  font-size: 16px;
  padding: 8px;
  border-radius: 3px;
  background-color: #2ace7e;
  border-color: #2ace7e;
  color: #ffffff;
}

.c0:focus {
  outline-color: #6A737B;
}

.c0:hover {
  background-color: #26b971;
  border-color: #26b971;
}

<Button
  disabled={false}
  fluid={false}
  loading={false}
  mode="primary"
  onClick={[Function]}
  size="regular"
  value="button"
>
  <styled.button
    disabled={false}
    fluid={false}
    mode="primary"
    onClick={[Function]}
    size="regular"
  >
    <StyledComponent
      disabled={false}
      fluid={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: inline-flex;
  border-width: 1px;
  border-style: solid;
  cursor: pointer;
  outline: 1px dotted transparent;
  outline-offset: 2px;
  justify-content: center;
  align-items: center;
  transition: 0.3s border-color, 0.3s background-color, 0.3s color, 0.3s outline-color;

  &:focus {
    outline-color: ",
              "#6A737B",
              ";
  }

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      mode="primary"
      onClick={[Function]}
      size="regular"
    >
      <button
        className="c0"
        disabled={false}
        mode="primary"
        onClick={[Function]}
        size="regular"
      >
        button
      </button>
    </StyledComponent>
  </styled.button>
</Button>
`;
